cmd/manager/main.go

```diff
@@ -50,6 +50,20 @@ func main() {
 	if err != nil {
 		log.Fatalf("failed to get watch namespace: %v", err)
 	}
+	// Create a new status reporter.
+	coWriter, err := operatorstatus.NewWriter()
+	if err != nil {
+		log.Fatalf("Error creating ClusterOpreator writer: %v", err)
+	}
+
+	// Report that the operator is progressing towards a new version.
+	err = operatorstatus.ReportProgressing(coWriter, namespace)
+	if err != nil {
+		log.Error(err)
+	}
+
+	// Start reporting status as a result of the number processed of syncs.
+	go operatorstatus.MonitorSyncs(coWriter, namespace)
```

Comment:
We could - but OpenShift is supposed to be self-recovering. If the CO API is not available when the operator starts, Marketplace should start reporting when it becomes available.

Reply:
Sure it can be self-recovering but there are ways to figure out if you are on an OpenShift vs a vanilla kube cluster like checking for the `ClusterVersion` API. Moreover I don't see how the marketplace operator comes up before CVO that owns this CRD.

Reply:
Or you make a command line option that you pass to the operator that indicates whether you report status or not.

Reply:
And this option will only be present in the OpenShift deployment file and not in the upstream one.